GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -2663,10 +2666,7 @@ private void SubmoduleStatusProvider_StatusUpdating(object sender, EventArgs e)
 
         private void SubmoduleStatusProvider_StatusUpdated(object sender, SubmoduleStatusEventArgs e)
         {
-            ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
-            {
-                await PopulateToolbarAsync(e.Info, e.Token);
-            });
+            e.Tasks.Add(PopulateToolbarAsync(e.Info, e.Token));
```

Comment:
❗️ This is a violation of the threading rules. A task which is not awaited in the context where it is created must be _joined_, not awaited (i.e. must go through `JoinableTaskFactory.RunAsync`).

Reply:
The idea is that the event starts tasks, collects them and right after that it awaits them all, so all the tasks are awaited in the same context where they were created (event is called synchronously). There was a question where I asked for your help https://github.com/gitextensions/gitextensions/pull/5617#discussion_r260778837